packages/gatsby-plugin-utils/src/validate.ts

```diff
@@ -0,0 +1,31 @@
+import { BaseValidationOptions, Schema } from "joi"
+
+const validationOptions: BaseValidationOptions = {
+  // Show all errors at once, rather than only the first one every time
+  abortEarly: false,
+  cache: true,
+  // Strip unknown values, this makes sure plugin authors define an extensive schema
+  // TODO: Discuss whether we want this, maybe not a great developer experience...
+  stripUnknown: true,
```

Comment:
Could you explain more exactly what this does?

Reply:
I misunderstood how this works, thanks for the clarification. Here's an example:

```JS
const schema = Joi.object({
  str: Joi.string()
})

const options = validatePluginOptions(schema, {
  str: `bla`,
  notInSchema: true
});

// With stripUnknown: false
Error: 'notInSchema' is not allowed

// With stripUnknown: true
console.log(options) // => { str: `bla` }
```

I think we actually want the former now 😅